docs/architecture.md

```diff
@@ -23,67 +23,99 @@
 
-TBC.
+A number of protocols come packages with the AEA framework.
 
-### Envelope
+* `oef`
+* `default`
+* `fipa`
 
-The `Envelope` is the core object which agents use to communicate with each other. An `Envelope` has four attributes:
+### Connection
 
-* `to`: defines the destination address
+A connection wraps an external SDK or API and manages the messaging. It allows the agent to connect to an external service which has a Python SDK or API. 
 
-* `sender`: defines the sender address
+The module `connections/base.py` contains two abstract classes which define a `Channel` and a `Connection`. A `Connection` contains one `Channel`.
 
-* `protocol_id`: defines the protocol_id
+The framework provides a number of default connections.
 
-* `message`: is a `bytes` field to hold the message in serialized form.
+* `local`: implements a local node.
+* `oef`: wraps the OEF SDK.
 
+### Skill
 
-### MainLoop
+Skills deliver economic value to the AEA by allowing an agent to encapsulate and call any kind of code. It encapsulates Handlers, Behaviours, and Tasks.
 
-TBC.
+* Handler: each skill has a single Handler which is responsible for the registered protocol messaging. Handlers implement reactive behaviour; by understanding the requirements contained in Envelopes, the Handler reacts appropriately to message requests. 
+* Behaviour: one or more Behaviours encapsulate sequences of actions that cause interactions with other agents initiated by the framework. Behaviours implement proactive behaviour.
+* Task: one or more Tasks encapsulate background work internal to the agent.
 
-### ProtocolMessage
 
-TBC. 
 
-### Director
+## Agent 
 
-TBC. 
+### Main loop
 
-### Orchestrator
+The `_run_main_loop()` function in the `Agent` class performs a series of activities while the `Agent` state is not stopped.
 
-TBC.
+* `act()`: this function calls the `act()` function of all registered Behaviours.
+* `react()`: this function grabs all Envelopes waiting in the InBox queue and calls the `handle()` function on them.
+* `update()`: this function loops through all the Tasks and executes them.
 
 ## Resources 
-### HanderRegistry 
-TBC.
-### BehaviourRegistry
-TBC.
-### TaskRegistry
-TBC.
-### ProtocolRegistry
-TBC.
+
+Registries hold Resources. There is one Registry for each type of Resource. The specific classes are in the `registries/base.py` module.
+
+* ProtocolRegistry.
+* HandlerRegistry. 
+* BehaviourRegistry.
+* TaskRegistry.
+
+
+## Director
+
+!!! TODO 
+
+## Filter
+
+!!! TODO 
+
 
 
 
 ## File structure
 
-An agent is structured in a directory with a configuration file, a directory with skills, a directory with protocols, a directory with connections and a main logic file that is used when running aea run.
+The file structure of an agent is fixed.
+
+The top level directory has the agent's name. Below is a `yaml` configuration file, then directories containing the connections, protocols, and skills, and a security certification file.
```

Comment:
What is the security certification file?

Reply:
I saw a `.pem` file dropped into the top level directory when I built an agent.

Reply:
@katharine-murphy ok this is the file containing the private key of the agent. So maybe call it that :)